src/types/data.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * 0x prefixed 20 bytes ethereum address encoded as hex string
+ */
+export type Address = string;
+
+/**
+ * Integer value, used to represent whole numbers without fractional parts
+ */
+export type Integer = number;
```

Comment:
I kind of like these types for semantic value, but not sure if they're worth it 🤔 Do consumers need to coerce everything to `Integer` to not get compile errors?

Reply:
no this is just syntactic sugar for 'number' type. so it will not generate compile errors.

this is simply to communicate clearly to consumers what exactly we're expecting